details/k8s/k8s_util.go

```diff
@@ -8,32 +8,35 @@ import (
 	"context"
 	"github.com/go-logr/logr"
 	"github.com/pkg/errors"
-	"k8s.io/apimachinery/pkg/types"
-
 	rbacv1 "k8s.io/api/rbac/v1"
 	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/kubernetes"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/NetApp-Polaris/astra-connector-operator/util"
 )
 
 type K8sUtil struct {
-	Client client.Client
-	log    logr.Logger
+	client.Client
+	kubernetes.Interface
```

Comment:
Can we not use https://github.com/NetApp/astra-connector-operator/blob/main/details/k8s/k8s_util.go#L94
to check and see if a CRD exists?

Reply:
The current logic to check for AKS is to look for a ClusterRole with the name aks-service. Do you know of a CRD we could use instead of that?

Reply:
👍